crypto/evp/p_rsa.c

```diff
@@ -104,11 +104,12 @@ static int pkey_ctx_is_pss(EVP_PKEY_CTX *ctx) {
 }
 
 // This method checks if the NID of |s_md| is the same as the NID of |k_md| when
-// |pkey_ctx_is_pss(ctx)| is true.
-static int pss_hash_algorithm_match(EVP_PKEY_CTX *ctx, const EVP_MD *k_md,
-                               const EVP_MD *s_md) {
-  if (pkey_ctx_is_pss(ctx) && k_md) {
-    if (s_md) {
+// |pkey_ctx_is_pss(ctx)| is true and there is PSS restriction, which means
```

Comment:
Nit: not totally clear for me what this description describes. Is this all cases for which the function returns true?

Reply:
The description says if `ctx` is pss and if the pss has restriction (specifying params).

> Is this all cases for which the function returns true?

This is just the condition when the algorithm match check should be performed. Not the condition when this function will returns true.